### DISCUSSION TITLE
Snipe-It V7 news?

### DISCUSSION THREAD

**Discussion by victorhooi** [snipe/snipe-it#14682] [Q&A] [answered]
Does anybody know what's the latest news on Snipe-It version 7? =)

I saw there was a blog post towards the end of last year about the upcoming Snipe-It V7 release:

https://grokstar.dev/news/snipeit-releases/2023/11/snipe-it-v7-roadmap/

which seems pretty exciting!

However, I couldn't find any moire recent news after that.

Does anybody know what happened since then?

Also - sorry if this is a stupid question - but is there anywhere we can follow the development of Snipe-It v7?

In the main [snipe-it Githnub repo](https://github.com/snipe/snipe-it), I saw there's a [`snipeit_v7` branch](https://github.com/snipe/snipe-it/tree/snipeit_v7). However, the last commit there seems to be from around 7 months ago.

There's also a [`snipe-it-dev` repo](https://github.com/grokability/snipe-it-dev) under the Grokability org, but that's pretty old as well.

Or is there somewhere we can see a demo of the v7 branch?

On the [demo page](https://snipeitapp.com/demo/) it says:
> The stable Snipe-IT demo is available at [demo.snipeitapp.com](https://demo.snipeitapp.com/).
To explore new features and bugfixes on the develop branch, we also provide a [develop demo](https://develop.snipeitapp.com/).

However, both of the stable and develop demo links seem to point to v6.3.4.

**Comment by snipe** [comment]
Great question! We had planned on releasing before the end of the year, but staff vacations, holidays, some unexpected illnesses, and such made that impossible. (Most of the discussion and transparency has been happing in our Discord channel, which is why there hasn’t been much here - also my fault for not adding another blog post for explaining the delay.)

Our current plans are: release Snipe-IT v6.4 this coming week, which will be the last of the 6.x line. Then merge the v7 branch into develop (which will make it available to the public for tire-kicking, etc.) We’ll also continue testing it internally as well, obviously. This puts us in a position where any new critical fixes will have to be hot fixed on master, instead of pushed into develop and then merged into master, then into a release. It’s a clunky place for us to be, and we don’t like to stay there very long, so we’ll try to keep that time down to a minimum, hopefully a week or so.

There were a lot of big framework level changes we had to make for v7, all while also continuing to add smaller features and fix bugs on v6.3, so these things never go exactly according to plan - at least not for us. A community contribution can introduce an unexpected issue, a previously undetected bug can sneak in, and there re only 24 hours in the day. But it’s still very much at the top of our priority list, and a lot of the big things we want to do once v7 is out. Even just constantly merging v6 features and bug fixes into v7 has meant constant merge conflicts and resolutions (poor @uberbrady), but we also have some v7 PRs that will be easier to test once it’s on develop.

**Comment by uberbrady** [comment, marked as answer]
Ah! Wrong branch. The one you want is: [snipeit_v7_laravel10](https://github.com/snipe/snipe-it/tree/snipeit_v7_laravel10) which has pretty recent changes in it. (The reason for the second branch name is the original plan for v7 wasn't going to have us move up to Laravel v10, but after some discussions we decided we *were* going to go that way, after all.) So that's the one you want to look at. In fact, I probably ought to just delete the old v7 branch because it's confusing.

**Reply by uberbrady** [reply]
I got a little nervous about deleting the old v7 branch since the 'real' branch still depends on some of those changes, so I just renamed it instead to [snipeit_v7_STALE](https://github.com/snipe/snipe-it/tree/snipeit_v7_STALE) which will hopefully be a little clearer.
